app/scripts/modules/amazon/src/serverGroup/configure/wizard/pages/ServerGroupBasicSettings.tsx

```diff
@@ -398,47 +294,16 @@ class ServerGroupBasicSettingsImpl extends React.Component<
             <div className="col-md-3 sm-label-right">
               Image <HelpField id="aws.serverGroup.imageName" />
             </div>
-            {values.viewState.useAllImageSelection && (
-              <div className="col-md-9">
-                <TetheredSelect
-                  clearable={false}
-                  placeholder="Search for an image..."
-                  required={true}
-                  valueKey="imageName"
-                  filterOptions={isLoadingImages ? (false as any) : undefined}
-                  options={images}
-                  optionRenderer={this.imageOptionRenderer}
-                  onInputChange={value => this.searchImages(value)}
-                  onChange={(value: Option<string>) => this.imageChanged(value)}
-                  onSelectResetsInput={false}
-                  onBlurResetsInput={false}
-                  onCloseResetsInput={false}
-                  value={selectedImage}
-                  valueRenderer={this.imageOptionRenderer}
-                />
-              </div>
-            )}
-            {!values.viewState.useAllImageSelection && (
-              <div className="col-md-9">
-                <TetheredSelect
-                  clearable={false}
-                  placeholder="Pick an image"
-                  required={true}
-                  valueKey="imageName"
-                  options={images}
-                  optionRenderer={this.imageOptionRenderer}
-                  onChange={(value: Option<string>) => this.imageChanged(value)}
-                  onSelectResetsInput={false}
-                  onBlurResetsInput={false}
-                  onCloseResetsInput={false}
-                  value={selectedImage}
-                  valueRenderer={this.imageOptionRenderer}
-                />
-                <a className="clickable" onClick={this.enableAllImageSearch}>
-                  Search All Images
-                </a>{' '}
-                <HelpField id="aws.serverGroup.allImages" />
-              </div>
+            {isExpressionLanguage(values.amiName) ? (
```

Comment:
Should `AmazonImageSelectInput` deal with this itself?

Reply:
I think this sort of logic belongs in `FormikFormField`/`FormField`
It's not there yet, but that's where I think it goes.